### PR TITLE
docs: Add PostCSS/Autoprefixer/CSSNano

### PR DESCRIPTION
**Pull request by nschonni**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

#### What changes did you make? 

1. Force added the compiled CSS in the `_site/assets/css` directory, without compression to show the diffs while the changes were being made.
1. Added PostCSS/Autoprefixer to add vendor prefixed values based on `last 2 browser releases and IE11` supported propteries.
1. Let Stylelint strip out the current manual brower prefixed properties, and show the noop with Autoprefixer adding them back in.
1. Add back minification through CSSNano
1. Cleanup the debugging and checked in complied files

#### Is there anything you'd like reviewers to focus on?

1. Each commit includes the checked in CSS so you can see the impact on adding it to the output, but I cleaned it out so you won't see it on the files tab of the PR
1. Couldn't currently add a `watch:postcss` because of https://github.com/postcss/postcss-cli/issues/388. Could probalby be worked around by moving the original compile target back to `src` and then letting PostCSS do the copy to `_site`.
1. This doesn't fix the SCSS being copied to `_site` by eleventy through the passthrough

### CODE REVIEW

**Comment by nschonni** [comment]
There are also a few warnings coming from autoprefixer around using `start`

```
$ npm run build:postcss

> docs-eslint@8.26.0 build:postcss
> postcss _site/assets/css --replace --verbose

Processing _site\assets\css\carbon-ads.css...
Processing _site\assets\css\docs-footer.css...
Processing _site\assets\css\docs-header.css...
Processing _site\assets\css\docs.css...
Processing _site\assets\css\eslint-site-header.css...
Processing _site\assets\css\forms.css...
Processing _site\assets\css\eslint-site-footer.css...
Processing _site\assets\css\foundations.css...
Processing _site\assets\css\languages.css...
Processing _site\assets\css\print.css...
Processing _site\assets\css\syntax-highlighter.css...
Processing _site\assets\css\utilities.css...
Processing _site\assets\css\components\alert.css...
Processing _site\assets\css\versions.css...
Processing _site\assets\css\components\buttons.css...
Processing _site\assets\css\components\docs-index.css...
Processing _site\assets\css\components\docs-navigation.css...
Processing _site\assets\css\components\hero.css...
Processing _site\assets\css\styles.css...
Processing _site\assets\css\components\index.css...
Processing _site\assets\css\components\language-switcher.css...
Processing _site\assets\css\components\resources.css...
Processing _site\assets\css\components\rules.css...
Processing _site\assets\css\components\social-icons.css...
Processing _site\assets\css\components\search.css...
Processing _site\assets\css\components\toc.css...
Processing _site\assets\css\components\tabs.css...
Processing _site\assets\css\components\version-switcher.css...
Processing _site\assets\css\components\theme-switcher.css...
Processing _site\assets\css\tokens\spacing.css...
Processing _site\assets\css\tokens\themes.css...
Processing _site\assets\css\tokens\typography.css...
Processing _site\assets\css\tokens\ui.css...
Finished _site\assets\css\eslint-site-header.css in 2.37 s
Finished _site\assets\css\docs-header.css in 2.38 s
Finished _site\assets\css\docs-footer.css in 2.38 s
Finished _site\assets\css\forms.css in 2.37 s
Finished _site\assets\css\carbon-ads.css in 2.39 s
Finished _site\assets\css\eslint-site-footer.css in 2.37 s
Finished _site\assets\css\docs.css in 2.38 s
Finished _site\assets\css\print.css in 2.37 s
Finished _site\assets\css\syntax-highlighter.css in 2.37 s
Finished _site\assets\css\components\alert.css in 2.37 s
Finished _site\assets\css\versions.css in 2.37 s
Finished _site\assets\css\foundations.css in 2.37 s
Finished _site\assets\css\utilities.css in 2.37 s
Finished _site\assets\css\languages.css in 2.37 s
Finished _site\assets\css\components\hero.css in 2.36 s
Finished _site\assets\css\components\docs-index.css in 2.37 s
Finished _site\assets\css\components\docs-navigation.css in 2.36 s
Finished _site\assets\css\components\buttons.css in 2.37 s
Finished _site\assets\css\components\search.css in 2.36 s
Finished _site\assets\css\components\language-switcher.css in 2.36 s
Finished _site\assets\css\styles.css in 2.36 s

2097:3  ⚠  start value has mixed support, consider using flex-start instead [autoprefixer]
2573:3  ⚠  start value has mixed support, consider using flex-start instead [autoprefixer]

Finished _site\assets\css\components\tabs.css in 2.36 s

9:3     ⚠  start value has mixed support, consider using flex-start instead [autoprefixer]

Finished _site\assets\css\components\resources.css in 2.37 s
Finished _site\assets\css\components\index.css in 2.37 s
Finished _site\assets\css\components\version-switcher.css in 2.36 s
Finished _site\assets\css\components\toc.css in 2.36 s
Finished _site\assets\css\components\social-icons.css in 2.36 s
Finished _site\assets\css\tokens\themes.css in 2.36 s
Finished _site\assets\css\tokens\typography.css in 2.36 s
Finished _site\assets\css\components\rules.css in 2.37 s

151:3   ⚠  start value has mixed support, consider using flex-start instead [autoprefixer]

Finished _site\assets\css\tokens\ui.css in 2.36 s
Finished _site\assets\css\tokens\spacing.css in 2.36 s
Finished _site\assets\css\components\theme-switcher.css in 2.36 s
```

**Review by snitin315 — COMMENTED** [review]
I would like @eslint/website-team to review it.

**Review by nzakas — APPROVED** [review]
LGTM. Would like someone else to double-check before merging.

File: docs/package.json
Comment (snitin315) [inline docs/package.json:15]: Maybe keep it at `src/assets/css` since we copy these assets to `_site`?

https://github.com/eslint/eslint/blob/e94a4a95ee301b0344d3292c37a0b29d8e18ab30/docs/.eleventy.js#L342

Comment (nschonni) [reply docs/package.json:15]: Yes, if you look in the build folder, the scss files are always copied. I tried looking at a way to fix it before, but I'm not familiar enough with eleventy

File: docs/package.json
Comment (nschonni) [inline docs/package.json:13]: Could potentially undo this, and then use the src/assets/css as the watch target for postcss

File: docs/package.json
Comment (harish-sethuraman) [inline docs/package.json:16]: Why have we considered removing the `--style=compressed` in building sass? and adding cssnano instead?

Comment (nschonni) [reply docs/package.json:16]: I explain that in the PR description

File: docs/package.json
Comment (harish-sethuraman) [reply docs/package.json:13]: We can have it like you have written. This is better. We can reduce writing and copying action to just directly writing to `_site`. Just confirm if changes work fine in dev env

**Review by harish-sethuraman — APPROVED** [review]
LGTM. Great work!

File: docs/package.json
Comment (bmish) [inline docs/package.json:61]: Do we still need to support IE 11 at this point? I would strongly advocate against explicitly supporting it because:

- It is [end-of-life](https://blogs.windows.com/windowsexperience/2022/06/15/internet-explorer-11-has-retired-and-is-officially-out-of-support-what-you-need-to-know) as of earlier this year.
- Actively supporting it increases maintenance/support burden and potentially imposes development limitations.
- Typically, IE support would be more important for a consumer-facing site like Amazon where some fraction of potential customers could still be using very outdated software and the site has no control over their software. ESLint's site is largely developer-facing, and it's hard to imagine any ESLint users still using IE 11, especially considering ESLint has explicit requirements that users must be using a very modern version of Node. I don't see how anyone with a modern Node environment could still be depending on IE.

Comment (nschonni) [reply docs/package.json:61]: I agree, but I added it based on the discussion in an earlier PR around removing fallbacks https://github.com/eslint/eslint/pull/16466#discussion_r1008641497

Comment (bmish) [reply docs/package.json:61]: It looks like previous discussion just assumed we wanted to maintain the status quo. I'm fine with maintaining the status quo in this PR, but my recommendation to stop explicitly supporting IE still stands (could be a good follow-up PR to consider) (CC: @nzakas).

File: docs/.eleventy.js
Comment (nschonni) [inline docs/.eleventy.js:465]: This one still doesn't seem to reload on the CSS changes still

File: docs/package.json
Comment (nschonni) [reply docs/package.json:15]: Went back and adjusted it so it chains through for the built CSS, but the SCSS is no longer copied through

File: docs/.eleventy.js
Comment (harish-sethuraman) [reply docs/.eleventy.js:465]: We tried to do the same thing with playground but it eleventy didn't detect file changes inside `_site` directory. Try with the postcss output in the same place `/assets/css` rather than inside the  `_site`. Eleventy detects the changes inside `css` directory and copies the file changes into `_site`.


Also you seem to use the `build:postcss` command for start script? In that case the postcss command is executed only once and when the scss file changes the output of `postcss` will be over written by that of `scss`?

Comment (nschonni) [reply docs/.eleventy.js:465]: I added the `watch:postcss` script, so the `npm run start`  also watches for the SCSS file changes, builds, then runs PostCSS

Comment (harish-sethuraman) [reply docs/.eleventy.js:465]: `"watch:postcss": "postcss src/assets/css -d src/assets/css --watch --poll"`

> Try with the postcss output in the same place /assets/css rather than inside the _site. Eleventy detects the changes inside css directory and copies the file changes into _site.


Even though we mention the `files: "./_site/css/**/*.css"` it doesnt detect 😞

File: docs/package.json
Comment (nzakas) [reply docs/package.json:62]: I generally feel like if we can get it for free, then let’s do it. In this case it seems like we can get IE11 support for free, so I don’t see any harm in keeping it.

**Comment by nzakas** [comment]
Where are we on this PR now? What are the next steps?

**Comment by harish-sethuraman** [comment]
we are stuck at css not reloading on dev server when changes occur. Other things are fine.

**Comment by nzakas** [comment]
Can we get unstuck? :) Is someone looking this?

**Comment by harish-sethuraman** [comment]
Yep. Suggested to https://github.com/eslint/eslint/pull/16502#discussion_r1020882707 this should most probably work :)

**Comment by harish-sethuraman** [comment]
Hey @nschonni sorry for the mention but wanted to confirm if you are still working on this? We can take it to completion :)

**Comment by nschonni** [comment]
@harish-sethuraman if you've got an idea for the problem with the watch, go for it. I think even with the watch problem, this probably is OK to land, as it only affects live development, not the end result

**Comment by harish-sethuraman** [comment]
So it works if I change `"watch:postcss": "postcss src/assets/css -d _site/assets/css --watch --poll",` to `"watch:postcss": "postcss src/assets/css -d src/assets/css --watch --poll",` (changed `_site` to `src`) in watch:postcss and build:postcss as Eleventy doesn't detect changes inside `_site`. And also added a passthrough for `assets/css` folder as well and hotreloading started working fine. I'm not able to push the changes as I don't have write access to your repo 😅 . Can you try this change and push it we can merge this 👍🏻

**Comment by nzakas** [comment]
I think the goal is that we want the same behavior that we have before this PR lands, so watching should work the same way.

Note that there is a conflict on this PR as well.

**Comment by nzakas** [comment]
Okay everyone, it's time to decide what to do with this PR. :) Are we moving forward with more fixes or closing?

**Comment by harish-sethuraman** [comment]
@nschonni you wanna take this up and complete? Its 95% done you can read more on the last 5% 👇🏻 

> So it works if I change "watch:postcss": "postcss src/assets/css -d _site/assets/css --watch --poll", to "watch:postcss": "postcss src/assets/css -d src/assets/css --watch --poll", (changed _site to src) in watch:postcss and build:postcss as Eleventy doesn't detect changes inside _site. And also added a passthrough for assets/css folder as well and hotreloading started working fine. I'm not able to push the changes as I don't have write access to your repo 😅 . Can you try this change and push it we can merge this 👍🏻

**Comment by nschonni** [comment]
I reverted back to old behaviour of copying the SCSS files to the _site, and used the recommended old folder locations. There may be race conditions now with the parallel watching, but I think i've reached the amount of time i'm willing to spend on this. Feel free to close, or push other changes, as "Allow edits from maintainers" has been on since the beginning @harish-sethuraman

**Comment by snitin315** [comment]
I'll finish this PR 👍🏻 .

**Comment by snitin315** [comment]
@harish-sethuraman your suggestion in https://github.com/eslint/eslint/pull/16502#issuecomment-1349677771 has been addressed, can you review once again?

**Review by snitin315 — APPROVED** [review]
LGTM, waiting for review from @harish-sethuraman

**Comment by harish-sethuraman** [comment]
LGTM, Thanks for helping with this 🙌🏻

**Pull request merged**
